src/test/java/io/github/sskorol/testcases/XlsxDataSupplierTests.java

```diff
@@ -29,4 +34,9 @@ public void shouldReadLocalExcelSpreadsheetWithoutSheet(final PersonWithoutSheet
     public void shouldReadLocalExcelSpreadsheetWithSheet(final PersonWithSheet person) {
         // not implemented
     }
+
+    @Test(dataProvider = "getDataWithCustomConverters")
```

Comment:
Hi @sskorol 

The unit test I added is failing because it could not take into consideration of my custom converters present in the src/test/java/io/github/sskorol/converters package.

Could you please advise on how to fix this. 

Apart from this all other issues are fixed.

Reply:
@amuthansakthivel are you getting some specific error? Can you share it?

Reply:
Common suite > DataSupplier tests > io.github.sskorol.testcases.DataSupplierTests > 

xlsxDataSupplierTestsShouldWork FAILED
    java.lang.AssertionError: 
    Expecting:
     <["shouldReadLocalExcelSpreadsheetWithCustomConverters(TestData(testCase=testcase1, browser=chrome, firstName=abcd, isFTE=true, age=20, phoneNumber=9876543210, list=null))",
        "shouldReadLocalExcelSpreadsheetWithCustomConverters(TestData(testCase=testcase1, browser=edge, firstName=dfgdg, isFTE=false, age=23, phoneNumber=+919876543210, list=null))",
        "shouldReadLocalExcelSpreadsheetWithCustomConverters(TestData(testCase=testcase2, browser=chrome, firstName=efgh, isFTE=true, age=21, phoneNumber=9876543210, list=null))",
        "shouldReadLocalExcelSpreadsheetWithCustomConverters(TestData(testCase=testcase3, browser=chrome, firstName=ijkl, isFTE=true, age=22, phoneNumber=9876543210, list=null))",
        "shouldReadLocalExcelSpreadsheetWithSheet(PersonWithSheet(row=1, firstName=Zikani, lastName=Nyirenda, dateOfBirth=1993-04-03))",
        "shouldReadLocalExcelSpreadsheetWithSheet(PersonWithSheet(row=2, firstName=Andrew, lastName=Mfune, dateOfBirth=1994-05-21))",
        "shouldReadLocalExcelSpreadsheetWithSheet(PersonWithSheet(row=4, firstName=Blessings, lastName=Mwafulirwa, dateOfBirth=1988-02-25))",
        "shouldReadLocalExcelSpreadsheetWithSheet(PersonWithSheet(row=6, firstName=Kondwani , lastName=Chikhula, dateOfBirth=1986-04-05))",
        "shouldReadLocalExcelSpreadsheetWithSheet(PersonWithSheet(row=8, firstName=Moses, lastName=Mpulula, dateOfBirth=1992-01-10))",
        "shouldReadLocalExcelSpreadsheetWithoutSheet(PersonWithoutSheet(row=1, firstName=Zikani, lastName=Nyirenda, dateOfBirth=1993-04-03))",
        "shouldReadLocalExcelSpreadsheetWithoutSheet(PersonWithoutSheet(row=2, firstName=Andrew, lastName=Mfune, dateOfBirth=1994-05-21))",
        "shouldReadLocalExcelSpreadsheetWithoutSheet(PersonWithoutSheet(row=4, firstName=Blessings, lastName=Mwafulirwa, dateOfBirth=1988-02-25))",
        "shouldReadLocalExcelSpreadsheetWithoutSheet(PersonWithoutSheet(row=6, firstName=Kondwani , lastName=Chikhula, dateOfBirth=1986-04-05))",
        "shouldReadLocalExcelSpreadsheetWithoutSheet(PersonWithoutSheet(row=8, firstName=Moses, lastName=Mpulula, dateOfBirth=1992-01-10))"]>
    to contain:
     <["shouldReadLocalExcelSpreadsheetWithoutSheet(PersonWithoutSheet(row=1, firstName=Zikani, lastName=Nyirenda, dateOfBirth=1993-04-03))",
        "shouldReadLocalExcelSpreadsheetWithoutSheet(PersonWithoutSheet(row=2, firstName=Andrew, lastName=Mfune, dateOfBirth=1994-05-21))",
        "shouldReadLocalExcelSpreadsheetWithoutSheet(PersonWithoutSheet(row=4, firstName=Blessings, lastName=Mwafulirwa, dateOfBirth=1988-02-25))",
        "shouldReadLocalExcelSpreadsheetWithoutSheet(PersonWithoutSheet(row=6, firstName=Kondwani , lastName=Chikhula, dateOfBirth=1986-04-05))",
        "shouldReadLocalExcelSpreadsheetWithoutSheet(PersonWithoutSheet(row=8, firstName=Moses, lastName=Mpulula, dateOfBirth=1992-01-10))",
        "shouldReadLocalExcelSpreadsheetWithSheet(PersonWithSheet(row=1, firstName=Zikani, lastName=Nyirenda, dateOfBirth=1993-04-03))",
        "shouldReadLocalExcelSpreadsheetWithSheet(PersonWithSheet(row=2, firstName=Andrew, lastName=Mfune, dateOfBirth=1994-05-21))",
        "shouldReadLocalExcelSpreadsheetWithSheet(PersonWithSheet(row=4, firstName=Blessings, lastName=Mwafulirwa, dateOfBirth=1988-02-25))",
        "shouldReadLocalExcelSpreadsheetWithSheet(PersonWithSheet(row=6, firstName=Kondwani , lastName=Chikhula, dateOfBirth=1986-04-05))",
        "shouldReadLocalExcelSpreadsheetWithSheet(PersonWithSheet(row=8, firstName=Moses, lastName=Mpulula, dateOfBirth=1992-01-10))",
        "shouldReadLocalExcelSpreadsheetWithCustomConverters(TestData(testCase=testcase1, browser=chrome, firstName=abcd, isFTE=true, age=20, phoneNumber=+919876543210, list=[selenium, appium, restassured]))",
        "shouldReadLocalExcelSpreadsheetWithCustomConverters(TestData(testCase=testcase1, browser=edge, firstName=dfgdg, isFTE=false, age=23, phoneNumber=+919876543210, list=[docker, k8s, openshift]))",
        "shouldReadLocalExcelSpreadsheetWithCustomConverters(TestData(testCase=testcase2, browser=chrome, firstName=efgh, isFTE=true, age=21, phoneNumber=+919876543210, list=[postman, soapui]))",
        "shouldReadLocalExcelSpreadsheetWithCustomConverters(TestData(testCase=testcase3, browser=chrome, firstName=ijkl, isFTE=true, age=22, phoneNumber=+919876543210, list=[automation, performance, security]))"]>
    but could not find:
     <["shouldReadLocalExcelSpreadsheetWithCustomConverters(TestData(testCase=testcase1, browser=chrome, firstName=abcd, isFTE=true, age=20, phoneNumber=+919876543210, list=[selenium, appium, restassured]))",
        "shouldReadLocalExcelSpreadsheetWithCustomConverters(TestData(testCase=testcase1, browser=edge, firstName=dfgdg, isFTE=false, age=23, phoneNumber=+919876543210, list=[docker, k8s, openshift]))",
        "shouldReadLocalExcelSpreadsheetWithCustomConverters(TestData(testCase=testcase2, browser=chrome, firstName=efgh, isFTE=true, age=21, phoneNumber=+919876543210, list=[postman, soapui]))",
        "shouldReadLocalExcelSpreadsheetWithCustomConverters(TestData(testCase=testcase3, browser=chrome, firstName=ijkl, isFTE=true, age=22, phoneNumber=+919876543210, list=[automation, performance, security]))"]>
        at io.github.sskorol.testdatasupplier/io.github.sskorol.testcases.DataSupplierTests.xlsxDataSupplierTestsShouldWork(DataSupplierTests.java:368)

30 tests completed, 1 failed

It is an assertion failure happened because the custom converters dint worked.

Reply:
Ok, l'll check it locally.

Reply:
@amuthansakthivel well, it took me a while to come back to this issue, as you may probably heard what's happening in Ukraine right now. Anyway, I debugged it a bit and seems like the root cause is [here](https://github.com/creditdatamw/zerocell/blob/master/core/src/main/java/com/creditdatamw/zerocell/handler/EntityExcelSheetHandler.java#L55), in a ZeroCell library, which prevents converters from being loaded suppresses the root cause. And the root cause is an attempt to reflectively access test-data-supplier classes. You may know that if you write your code in a modular style (Java 9+), you have to explicitly grant access to 3rd-party libs to your resources. Otherwise, you get an exception that some module can't read or export the other. The fix is pretty straightforward: we just need to export test-data-supplier package to zerocell.core. More time spent on root cause analysis. Ideally, public libs' owners should never suppress exceptions in such a way if it may break the key feature. Moreover, if you don't have a logger configured, you'll never catch the error at all.

Reply:
@amuthansakthivel you can add the following instruction to the test task or I can include all of your changes with a fix into my next PR:
```kotlin
"--add-exports", "${modulePrefix}.converters=zerocell.core"
```